### DISCUSSION TITLE
FATFS partition from Pycom cannot be mounted on official Micropython

### DISCUSSION THREAD

**Discussion by x4t-de** [micropython/micropython#11226] [ESP32] [answered]
Dear community,

First of all thanks to all of the contributers to this awesome project.

I am trying to bring the official Microypthon port of ESP32 to a Pycom module (WiPy). While all the basic things work, including SD card access, I have one problem: the flash partition is not recognized if it has been created under Pycom's firmware as FAT. I managed to do that with LittleFS, but with FAT there still seems to be a problem. I have som WiPy's with LittleFS and some with FAT for the data partition, and I would like to leave the data partition intact and do not have to reformat them.
I flashed Pycom's firmware with the "Pycom Firmware Update", chose to erased all flash and use FATFS. pycom.bootmgr() shows that the file system is FAT. I added a new file and wrote some text to it. The file is still there after a reset.
Then I flashed Micropython und performed the following commands on the REPL:
```
>>> from flashbdev import bdev
>>> bdev
<Partition type=1, subtype=129, address=4194304, size=4194304, label=vfs, encrypted=0>
>>> vfs = os.VfsFat(bdev)
>>> vfs
<VfsFat>
>>> os.mount(vfs,"/flash")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 19] ENODEV
>>> buf = bytearray(bdev.ioctl(5, 0))  # 5 is SEC_SIZE
>>> bdev.readblocks(0, buf)
>>> buf
bytearray(b'\xeb\xfe\x90MSDOS5.0\x00\x02\x04\x01\x00\x01\x00\x02\x00 \xf8\x07\x00?\x00\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80\x00)\x00\x00!\xecNO NAME    FAT     \x00
```

If I then make a new file system under Micropython (which works and I also can add and write to files) the beginning of the flash looks like this:
```
bytearray(b'\xeb\xfe\x90MSDOS5.0\x00\x10\x01\x01\x00\x01\x00\x02\x00\x04\xf8\x01\x00?\x00\xff\x00\x00\x00\x00\x00\x00\x00\x00\x00\x80\x00)\xc0\x00!(NO NAME    FAT     \x00\
```

Has anyone an idea of what might be the culprit? Or has anyone even tried this and succeeded? Help is very welcome, and I thank you all in advance :)

JR

**Comment by x4t-de** [comment]
Interesting enough, Pycom's Micropython version also does not recognize the filesystem that Micropython created. I also checked the source files of oofatfs. Git does not show any difference in the oofatfs lib files between Pycom and Micropython. So the difference must be somewhere else.

**Comment by x4t-de** [comment, marked as answer]
For somebody who maybe has the same problem: I did some more tests and investigations. This might be a noob error, but I didn't check all of the configurations that the oofatfs lib lets the user define. The main problem was the sector size definition. 
The filesystem maximum sector size `MICROPY_FATFS_MAX_SS` was undefined in Pycom's ESP32 port, which in ff.h results in min ss size == max ss size == 512. Micropython defines min == 512 and max == 4096. After changing the config to match the one from Pycom, FAT FS is recognized by Micropython port, mounted under "flash", and there are files listed in this directory when calling os.listdir("/flash").

Now I am facing the following problem: the files listed have names that come from the content of the files. E.g. one file is named "------BEG.IN" which is part of a certificate, and ENOENT error is thrown when I try to read from it. But I will post that as a different issue, because this one is about filesystem and mounting and solved.

New discussion: #12025

**Comment by brianredbeard** [comment]
FYI, @x4t-de, you didn't open a new issue but instead began a new discussion.

**Reply by x4t-de** [reply]
Thanks for pointing that out. Sorry for the wrong wording. I meant discussion, as I wasn't sure if that really is an issue of Micropython or just my own mistake. I'll correct it.
